src/domains/OptimismDomain.sol

```diff
@@ -84,40 +92,18 @@ contract OptimismDomain is BridgedDomain {
 
     function relayToHost(bool switchToHost) external override {
         hostDomain.selectFork();
+        address portal = L1MessengerLike(address(l1Messenger)).portal();
 
         // Read all L2 -> L1 messages and relay them under Primary fork
-        // Note: We bypass the L1 messenger relay here because it's easier to not have to generate valid state roots / merkle proofs
         Vm.Log[] memory logs = RecordedLogs.getLogs();
         for (; lastToHostLogIndex < logs.length; lastToHostLogIndex++) {
             Vm.Log memory log = logs[lastToHostLogIndex];
             if (log.topics[0] == SENT_MESSAGE_TOPIC && log.emitter == address(l2Messenger)) {
                 address target = address(uint160(uint256(log.topics[1])));
-                (address sender, bytes memory message,,) = abi.decode(log.data, (address, bytes, uint256, uint256));
-                // Set xDomainMessageSender
-                vm.store(
-                    address(l1Messenger),
-                    bytes32(uint256(204)),
-                    bytes32(uint256(uint160(sender)))
-                );
-                vm.startPrank(address(l1Messenger));
-                (bool success, bytes memory response) = target.call(message);
-                vm.stopPrank();
-                vm.store(
-                    address(l1Messenger),
-                    bytes32(uint256(204)),
-                    bytes32(uint256(0))
-                );
-                if (!success) {
-                    string memory rmessage;
-                    assembly {
-                        let size := mload(add(response, 0x44))
-                        rmessage := mload(0x40)
-                        mstore(rmessage, size)
-                        mstore(0x40, add(rmessage, and(add(add(size, 0x20), 0x1f), not(0x1f))))
-                        returndatacopy(add(rmessage, 0x20), 0x44, size)
-                    }
-                    revert(rmessage);
-                }
+                (address sender, bytes memory message, uint256 nonce, uint256 gasLimit) = abi.decode(log.data, (address, bytes, uint256, uint256));
+                stdstore.target(portal).sig("l2Sender()").checked_write(address(l2Messenger));
```

Comment:
I'm trying to understand why the `l2Messenger` is set here. If I look at the mainnet portal contract, in `l2Sender` it says the following: "Address of the L2 account which initiated a withdrawal in this transaction. If the of this variable is the default L2 sender address, then we are NOT inside of a call to finalizeWithdrawalTransaction.".

Is it because it is the default sender address?

Reply:
This is to satisfy [this check](https://github.com/ethereum-optimism/optimism/blob/3a62bccd6c5464891d0d6282264022d240d05b60/packages/contracts-bedrock/contracts/L1/L1CrossDomainMessenger.sol#L46) during relaying. Normally the l2Sender is set from its default value (some constant) to the l2Messenger by the portal [here](https://github.com/ethereum-optimism/optimism/blob/3a62bccd6c5464891d0d6282264022d240d05b60/packages/contracts-bedrock/contracts/L1/OptimismPortal.sol#L341).

Reply:
Understood, thanks.